src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11KeyStore.java

```diff
@@ -232,7 +231,7 @@ private static class PasswordCallbackHandler implements CallbackHandler {
         private PasswordCallbackHandler(char[] password) {
             if (password != null) {
                 this.password = password.clone();
-                Cleaner.create().register(this,
+                P11Util.cleaner.register(this,
                         () -> Arrays.fill(this.password, ' '));
```

Comment:
This lambda most probably capture `this` so it will create a leak.

Reply:
I was wondering to fix it in another PR as there is another issue in the block.  Better to fix it sooner.